### PR TITLE
WIP: add "clear row" to table body

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated Table story](https://deploy-preview-446--neo-react-library-storybook.netlify.app/?path=/story/components-table--editable-data)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

This work adds the "clear row" to the table. **Functionality**:
- If any rows are selected, the "clear row" should be visible.
- If no rows are selected, the "clear row" is not visible.
- If all rows are selected, the "clear row" should allow the user to clear the selection.
- If some rows but not all are selected, the "clear row" should allow the user to select all rows.

`@avaya-dux/dux-design`:

- check for consistency with [figma spec](https://www.figma.com/design/o62kJbRQpFq9T0fGwKvCMA/Component-Exploration-FY22---v3?node-id=3313-531108&t=gZyksnb7xnO0buSX-0)
- check for "clear row" functionality and confirm that it is a good UX
- test a11y

`@avaya-dux/dux-devs`:

- the usual
- test a11y
- this is a large PR due to adding and updating many tests, I've added notes in the tests for the new and updated functionality 
- the only functional changes are in the `TableBody.tsx` file
  - added a `ClearSelectionRow` component that displays when any rows are selected
  - added a `TableDataRows` component that is a copy-paste of the original component
  - I added these to make their parent component (`TableBody`) easier to understand

## Summary by CodeRabbit

- **New Features**
  - Enhanced table selection capabilities, including new options for clearing selections and selecting all items on the page or in the table.
  - Updated `TableHeader` with dynamic `aria-label` for improved accessibility.

- **Improvements**
  - Added buttons for clearing all selections directly within the table.
  - Improved multi-selection logic for better user experience across different pages.

- **Localization**
  - Introduced new translations for item selection messages to better support internationalization.

### CODE REVIEW

File: src/components/Table/Table.test.jsx [hunk 1]
```diff
@@ -218,7 +220,7 @@ describe("Table", () => {
 			);
 
 			const headerCheckbox = getByLabelText(
-				FilledFields.translations.header.selectAll,
+				FilledFields.translations.header.selectPage,
```
Comment (joe-s-avaya) [inline src/components/Table/Table.test.jsx:223]: **NOTE**: in my last PR I changed the functionality of the header checkbox but didn't update it's aria-label to reflect that change. I have update the aria-label, and thus I had to also update the tests to reflect the new wording.

File: src/components/Table/Table.test.jsx [hunk 2]
```diff
@@ -314,19 +316,112 @@ describe("Table", () => {
...
 
 			expect(screen.getByText("no data available")).toBeVisible();
 
-			expect(screen.getByLabelText(selectAllCheckboxLabel).checked).toBeFalsy();
+			expect(
+				screen.getByLabelText(selectPageCheckboxLabel).checked,
+			).toBeFalsy();
+		});
+
+		it("does not show 'clear-row' when no rows are selected", async () => {
```
Comment (joe-s-avaya) [inline src/components/Table/Table.test.jsx:337]: **NOTE**: this test and the below are the new tests

**Comment by Justin-Avaya** [comment]
Meeting with @matthewAvaya 
A number of issues still exist. Matt will reach out with clarifying questions.

**Review by yangAtSpoken — COMMENTED** [review]
First, perhaps a design question: Is it possible to make the clear row optional?  The functionality provided by clear row is the same as the menu.
Second: reuse the logic between the menu and clear row as the code look exactly the same to me.

**Comment by joe-s-avaya** [comment]
> First, perhaps a design question: Is it possible to make the clear row optional? The functionality provided by clear row is the same as the menu. Second: reuse the logic between the menu and clear row as the code look exactly the same to me.

1) good call, I'll talk to Matt about it
2) oh, yes, good call, will do

File: src/components/Table/TableComponents/TableBody.tsx [hunk 3]
```diff
@@ -15,24 +24,153 @@ import type { TableBodyProps } from "../types";
...
+						selectableRows={selectableRows}
+						translations={translations}
+					/>
+				</>
+			)}
+		</tbody>
+	);
+};
+
+const ClearSelectionRow: TableBodyComponentType = ({
```
Comment (enrique-prado) [inline src/components/Table/TableComponents/TableBody.tsx:61]: nit: This file is getting rather large... Should this new ClearSelectionRow be in its own file?

**Review by enrique-prado — APPROVED** [review]
LGTM, just the one nit.

File: src/components/Table/TableComponents/TableBody.tsx (hunk 3 above)
Comment (joe-s-avaya) [reply src/components/Table/TableComponents/TableBody.tsx:61]: ooh, good call, I will make that update in the next PR

**Comment by Justin-Avaya** [comment]
Just a few things around accessibility and the UX

**Accessibility**

While tabbing through the UI, i noticed the black focus box seemed offset. 
See examples below:

Example
Screenshot 2024-07-15 at 11 36 03 AM

Example
Screenshot 2024-07-15 at 11 34 13 AM

Example
Screenshot 2024-07-15 at 11 33 56 AM

<hr>

With the two rows selected in the below example, the bottom selected box seems to be bloated. This may be something caused by the select box being in a focused state, regardless I'm not sure it looks right.
See example below:

Example
Screenshot 2024-07-15 at 11 40 38 AM

<hr>

**UX**

While I like the concept of the blue banner, what seems to happen is a rather large jump when I select one of the rows with all other rows deselected. While playing, I found myself selecting and deselecting boxes unintentionally due to this jumping effect.

See example below:

Example

https://github.com/user-attachments/assets/a0619a8d-81fa-4653-928f-14bf2c74a4c3

<hr>

Since the PR has requested "check for "clear row" functionality and confirm that it is a good UX" I will close with the following:

With the blue banner displayed, I found myself not wanting the "Select all items in the table" rather, I kept wishing it was a "clear selected row(s) on page" instead.

My rational is if I want to select all items on all pages, I would not select a single row and then select the "Select all items in the table" from the blue banner. The same amount of clicks to select all items can be done by clicking the down chevron and then clicking "Select all items in the table" from the menu.

I can not find a quick and easy way to clear selected rows unless I select everything in the entire table and only then I am presented with an option to clear all items in the table whether that is from the blue banner or via the down chevron menu.

**Comment by joe-s-avaya** [comment]
Need to merge due to dev conflicts. Will open another PR with requested updates.

**Pull request merged**